spec/trick_bag/io/gitignore_spec.rb

```diff
@@ -7,70 +7,45 @@
 GI=TrickBag::Io::Gitignore
 
 describe TrickBag::Io::Gitignore do
-
-  # Pass this a block to perform in the test directory.
-  def create_and_populate_test_directory(ignore_spec)
-    Dir.mktmpdir do |tmpdir|
-      Dir.chdir(tmpdir) do
-        %w(a_dir  .hidden_dir  x/y/z).each { |dirspec| FileUtils.mkdir_p(dirspec) }
-        %w(root  a_dir/sub  .hidden_dir/.hidden_file  x/y/z/deep).each { |f| FileUtils.touch(f) }
-        File.write('.gitignore', ignore_spec.join("\n"))
-        yield
-      end
+  shared_examples_for 'testing a pair' do |ignore_spec, expected_result, filename|
```

Comment:
This is great stuff, thank you! In addition to cleaning up this test code, you have taught me how to use these RSpec features.